Include <sys/socket.h> before <net/if.h> to fix build failure on OS X < 10.9:

error: field has incomplete type 'struct sockaddr'

https://github.com/giampaolo/psutil/issues/2365
--- psutil/arch/osx/net.c.orig	2023-06-01 11:41:59.000000000 -0500
+++ psutil/arch/osx/net.c	2024-01-29 03:00:55.000000000 -0600
@@ -9,11 +9,11 @@
 // https://github.com/giampaolo/psutil/blame/efd7ed3/psutil/_psutil_osx.c
 
 #include <Python.h>
+#include <sys/socket.h> // must be included before <net/if.h>
 #include <net/if.h>
 #include <net/if_dl.h>
 #include <net/route.h>
 #include <sys/sysctl.h>
-#include <sys/socket.h>
 
 #include "../../_psutil_common.h"
 
